### DISCUSSION TITLE
Suddenly reloading page

### DISCUSSION THREAD

**Discussion by ahmad-azzam** [vercel/next.js#36441] [Help]
After a few minutes I did "next dev -p 3000" , the app sometimes reloads itself , why is that?
it also applies when my application is already in production

**Comment by icyJoseph** [comment]
Hi,

How is this happening? are you using any template? Is there any service worker installed on `localhost:3000` that may be crashing with the next app?

We need more information here. If you could make a reproduction repo, that'd be great. (I did start a new project, waited 10 seconds and nothing happened)

**Reply by ahmad-azzam** [reply]
My next version is 9.5.5, i use material-ui and not install service worker in this app

This is quite confusing to me because there is no error when I see the console, it's just that suddenly my application reloads itself

**Reply by icyJoseph** [reply]
Ah that's still not enough. I changed to next 9.5.5 with react 17, and still no issues. 

What do you mean with "reload"? Is it like calling `window.location.reload`?

**Reply by ahmad-azzam** [reply]
yaa , like `window.reload` ,

i have done command "`next build`" then "`node server`" , and this reloading problem doesn't appear anymore.. so what is the difference between " `next build` -> `node server` " and "`next dev -p 3000`" ?

**Reply by icyJoseph** [reply]
Are you using a custom server? `next build` is typically followed by `next start`, and `next dev` behaves as much as possible like them, but if there's a custom server then it becomes harder to tell what the problem might be. Not sure why you do `node server`.

The only thing I could see happening is that there's not reloading of the page, but some HMR (hot module reloading), not well handled is causing the SPA to remount. If you open your devTools, and in the console options set `preserve logs`, you should see a message saying `Navigated to <domain name>`:


<img width="1507" alt="Screenshot 2022-04-26 at 09 01 30" src="https://user-images.githubusercontent.com/21013447/165241220-6e5059e2-f13a-4c15-a13a-866a723dd056.png">


If that's the case, then theres' an actual `window.location.reload` call, AKA true page reload, otherwise, it is a HMR problem.

**Reply by timneutkens** [reply]
> My next version is 9.5.5

Next.js 12 and later does have this edge case where it reloads the page at random, so I'd recommend upgrading to the latest version of Next.js.
